packages/@react-spectrum/s2/src/SelectBox.tsx

```diff
@@ -0,0 +1,209 @@
+  width: {
+    default: {
+      size: {
+        XS: 100,
+        S: 128,
+        M: 136,
+        L: 160,
+        XL: 192
+      }
+    },
+    orientation: {
+      horizontal: {
+        size: {
+          XS: 'auto',
+          S: 'auto',
+          M: 'auto',
+          L: 'auto',
+          XL: 'auto'
+        }
+      }
+    }
+  },
+  
+  height: {
+    default: {
+      size: {
+        XS: 100,
+        S: 128,
+        M: 136,
+        L: 160,
+        XL: 192
+      }
+    },
+    orientation: {
+      horizontal: {
+        size: {
+          XS: 'auto',
+          S: 'auto',
+          M: 'auto',
+          L: 'auto',
+          XL: 'auto'
+        }
+      }
+    }
+  },
+  
+  minWidth: {
+    orientation: {
+      horizontal: 160
+    }
+  },
+  
+  maxWidth: {
+    orientation: {
+      horizontal: 272
+    }
+  },
+
+  padding: {
+    size: {
+      XS: 12,
+      S: 16,
+      M: 20,
+      L: 24,
+      XL: 28
+    }
+  },
+  
+  borderRadius: 'lg',
+  backgroundColor: 'layer-2',
+  boxShadow: {
+    default: 'emphasized',
+    isHovered: 'elevated',
+    isSelected: 'elevated',
+    forcedColors: 'none'
+  },
+  position: 'relative',
+  borderWidth: 2,
+  borderStyle: {
+    default: 'solid',
+    isSelected: 'solid'
+  },
+  borderColor: {
+    default: 'transparent',
+    isSelected: 'gray-900',
+    isFocusVisible: 'transparent'
+  },
+  transition: 'default'
+}, getAllowedOverrides());
+
+const checkboxContainer = style({
+  position: 'absolute',
+  top: 16,
+  left: 16
+}, getAllowedOverrides());
+
+/**
+ * SelectBox components allow users to select options from a list.
+ * They can behave as radio buttons (single selection) or checkboxes (multiple selection).
+ */
+export const SelectBox = /*#__PURE__*/ forwardRef(function SelectBox(props: SelectBoxProps, ref: FocusableRef<HTMLLabelElement>) {
+  [props, ref] = useSpectrumContextProps(props, ref, SelectBoxItemContext);
+  let {children, value, isDisabled = false, isSelected, onChange, UNSAFE_className = '', UNSAFE_style} = props;
+  let inputRef = useRef<HTMLInputElement | null>(null);
+  let domRef = useFocusableRef(ref, inputRef);
+  
+  let groupContext = useContext(SelectBoxContext);
+  let {
+    allowMultiSelect = false,
+    size = 'M',
+    orientation = 'vertical'
+  } = groupContext;
+
+  const Selector = allowMultiSelect ? AriaCheckbox : AriaRadio;
+  
+  // Handle controlled selection
+  const handleSelectionChange = (selected: boolean) => {
+    if (onChange) {
+      onChange(selected);
+    }
+  };
+  
+  return (
+    <Selector
+      value={value}
+      isDisabled={isDisabled}
+      isSelected={isSelected}
+      onChange={handleSelectionChange}
+      ref={domRef}
+      inputRef={inputRef}
+      style={UNSAFE_style}
+      className={renderProps => UNSAFE_className + selectBoxStyles({...renderProps, size, orientation}, props.styles)}>
+      {renderProps => (
+        <>
+          {(renderProps.isSelected || renderProps.isHovered) && (
```

Comment:
might also need to take into account `renderProps.isFocusVisible` for keyboard users